### PR TITLE
Fix bugs in unstake using controller address in polkadot

### LINKED ISSUE #4145: Bug in Unstake using a controller address in polkadot
**Describe the bug**

- When we sign in using a controller address and we want to unstake, it allows you to continue if the amount is 0, this should never happen. 
- Set max button does nothing when you want to unstake from a validator you are not actively staking.

**Expected behavior**

- Validation should fail in that case preventing user to continue.
- Set max button should set the total bonded amount in all cases.

### PR DESCRIPTION
**Pull request by mariopino**
Closes #4145

**Description:**

- [x] When we sign in using a controller address and we want to unstake, validation should fail if amount is 0.
- [x] Enable Set max button in Unstake modal even if you are at a validator page you are not actively staking in.
- [x] Enable Unstake button for all validator pages in polkadot.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by mariopino** [comment]
In polkadot unstake (aka unbond) is not really related with unstake from a single validator so I think we can also hide the From field.

**Comment by Bitcoinera** [comment]
Is this one good to go or still WIP?

**Comment by faboweb** [comment]
Unstake relates to a validator eventually if the user uses a controller account. That would be an unnomination.

File: src/ActionModal/components/UndelegationModal.vue [hunk 1]
```diff
@@ -450,7 +449,35 @@ export default {
         return !this.address
       },
     },
-
+    totalStaked: {
+      query: gql`
+        query overview($networkId: String!, $address: String!) {
```
Comment (faboweb) [inline src/ActionModal/components/UndelegationModal.vue:454]: please don't use the overview query. I would like to remove that query soon.

File: src/components/staking/PageValidator.vue [hunk 2]
```diff
@@ -79,7 +79,11 @@
         <TmBtn
           id="undelegation-btn"
           class="undelegation-btn"
-          :disabled="delegation.amount === 0 && !isInactiveValidator"
+          :disabled="
+            delegation.amount === 0 &&
+            !isInactiveValidator &&
```
Comment (faboweb) [inline src/components/staking/PageValidator.vue:84]: @Bitcoinera what is this again? I think we need to rename/refactor this. I can't read this.

**Comment by mariopino** [comment]
> Unstake relates to a validator eventually if the user uses a controller account. That would be an unnomination.

Agree! that flow is not very clear in polkadot, would be nice to discuss the best way to improve it.

File: src/components/staking/PageValidator.vue [hunk 3]
```diff
@@ -79,7 +79,11 @@
...
+            currentNetwork.network_type !== `polkadot`
```
Comment (faboweb) [inline src/components/staking/PageValidator.vue:85]: this doesn't work. case:
- polkadot
- none/controller account
- user has no delegation with a validator

File: src/ActionModal/components/UndelegationModal.vue [hunk 4]
```diff
@@ -193,11 +193,19 @@ export default {
...
-        ({ validator }) =>
-          validator.operatorAddress === this.sourceValidator.operatorAddress
-      )
-      return delegation ? Number(delegation.amount) : 0
+      if (this.currentNetwork.network_type === `polkadot`) {
+        const totalStaked = this.delegations.reduce(
+          (accum, delegation) =>
+            accum += parseFloat(delegation.amount)
+        , 0)
+        return totalStaked.toFixed(3) || 0
```
Comment (faboweb) [inline src/ActionModal/components/UndelegationModal.vue:201]:
```suggestion
        return totalStaked.toFixed(6) || 0
```

File: src/ActionModal/components/UndelegationModal.vue [hunk 5]
```diff
@@ -321,15 +329,10 @@ export default {
   validations() {
     return {
       amount: {
-        required: (x) =>
-          this.currentNetwork.network_type === "polkadot" || (!!x && x !== `0`),
+        required: (x) => !!x && x !== `0`,
```
Comment (faboweb) [inline src/ActionModal/components/UndelegationModal.vue:332]:
```suggestion
        required: (amount) => {
          // In Polkadot we don't need to unbond tokens, the user may just want to unnominate a validator
          // stash accounts can't do anything else but unbond so we make it required
          // none accounts can't access this modal
          if (
            this.currentNetwork.network_type === "polkadot" &&
            ["controller", "stash/controller"].includes(
              this.session.addressRole
            )
          ) {
            return true
          }
          return !!amount && amount !== `0`
        },
```

File: src/ActionModal/components/UndelegationModal.vue [hunk 6]
```diff
@@ -321,15 +329,10 @@ export default {
...
         decimal,
-        max: (x) =>
-          this.currentNetwork.network_type === "polkadot" ||
-          Number(x) <= this.maximum,
-        min: (x) =>
-          this.currentNetwork.network_type === "polkadot" ||
-          Number(x) >= SMALLEST,
+        max: (x) => Number(x) <= this.maximum,
```
Comment (faboweb) [inline src/ActionModal/components/UndelegationModal.vue:334]:
```suggestion
        max: (x) => {
            if (this.currentNetwork.network_type === `polkadot`) {
              return this.totalStaked || 0
            }
            return Number(x) <= this.maximum
          },
```

File: src/ActionModal/components/UndelegationModal.vue [hunk 7]
```diff
@@ -321,15 +329,23 @@ export default {
...
+        },
         decimal,
-        max: (x) =>
-          this.currentNetwork.network_type === "polkadot" ||
-          Number(x) <= this.maximum,
-        min: (x) =>
-          this.currentNetwork.network_type === "polkadot" ||
-          Number(x) >= SMALLEST,
+        max: (x) => Number(x) <= this.maximum,
+        min: (x) => Number(x) >= SMALLEST,
```
Comment (faboweb) [inline src/ActionModal/components/UndelegationModal.vue:348]:
```suggestion
        min: (x) => {
          // see required
          if (
            this.currentNetwork.network_type === "polkadot" &&
            ["controller", "stash/controller"].includes(
              this.session.addressRole
            )
          ) {
            return true
          }
          return Number(x) >= SMALLEST
        },
```

File: src/ActionModal/components/UndelegationModal.vue (hunk 1 above)
Comment (mariopino) [reply src/ActionModal/components/UndelegationModal.vue]: Thanks for replacing

**Pull request merged**